src/ctia/entity/investigation.clj

```diff
@@ -1,60 +1,38 @@
 (ns ctia.entity.investigation
-  (:require [ctia.domain.entities :refer [default-realize-fn]]
-            [ctia.store :refer :all]
-            [ctia.http.routes
-             [common :refer [BaseEntityFilterParams PagingParams SourcableEntityFilterParams]]
-             [crud :refer [entity-crud-routes]]]
-            [ctia.schemas
-             [utils :as csu]
-             [core :refer [def-stored-schema
-                           CTIAEntity]]
-             [sorting :as sorting]]
-            [ctia.schemas.graphql
-             [sorting :as graphql-sorting]
-             [flanders :as flanders]
-             [helpers :as g]
-             [pagination :as pagination]]
-            [ctia.stores.es
-             [mapping :as em]
-             [store :refer [def-es-store]]]
-            [ctim.schemas.investigation :as inv]
-            [flanders
-             [schema :as f-schema]
-             [spec :as f-spec]
-             [utils :as fu]]
-            [schema-tools.core :as st]
-            [schema.core :as s]
-            [ctia.schemas.graphql.ownership :as go]))
-
-(s/defschema Investigation
-  (st/merge (f-schema/->schema inv/Investigation)
-            CTIAEntity
-            {s/Keyword s/Any}))
-
-(f-spec/->spec inv/Investigation "investigation")
-
-(s/defschema PartialInvestigation
-  (st/merge (f-schema/->schema (fu/optionalize-all inv/Investigation))
-            CTIAEntity
-            {s/Keyword s/Any}))
-
-(s/defschema PartialInvestigationList
-  [PartialInvestigation])
-
-(s/defschema NewInvestigation
-  (st/merge (f-schema/->schema inv/NewInvestigation)
-            CTIAEntity
-            {s/Keyword s/Any}))
-
-(f-spec/->spec inv/NewInvestigation "new-investigation")
-
-(def-stored-schema StoredInvestigation Investigation)
-
-(s/defschema PartialStoredInvestigation
-  (csu/optional-keys-schema StoredInvestigation))
-
-(def realize-investigation
-  (default-realize-fn "investigation" NewInvestigation StoredInvestigation))
+  (:require
+   [ctia.http.routes
+    [common :refer [BaseEntityFilterParams
+                    PagingParams
+                    SourcableEntityFilterParams]]
+    [crud :refer [entity-crud-routes]]]
+   [ctia.stores.es
+    [mapping :as em]
+    [store :refer [def-es-store]]]
+   [ctia.entity.investigation.schemas
+    :refer [Investigation
+            PartialInvestigation
+            PartialStoredInvestigation
+            NewInvestigation
+            StoredInvestigation
+            PartialInvestigationList
+            realize-investigation]]
+   [schema-tools.core :as st]
+   [schema.core :as s]
+   [ctia.schemas.sorting :as sorting]))
+
+(def snapshot-action-fields-mapping
+  {:object_ids {:type "text"
+                :analyzer "text_analyzer"
+                :search_quote_analyzer "text_analyzer"
+                :search_analyzer "search_analyzer"
+                :include_in_all false}
+   :targets {:type "nested"
+             :include_in_all false}
```

Comment:
You should add a more specific mapping. If you do not do so, some fields mapping will be dynamically inferred and probably with a bad type (for instance, "2020-05-01" could be inferred as a string instead of a date).
https://github.com/threatgrid/ctia/blob/master/src/ctia/stores/es/mapping.clj#L227

Reply:
> You should add a more specific mapping.

I'll see what I can do but IIRC this is a bit of a junk drawer.

Reply:
Well, it seems that it is a [pretty well defined](https://github.com/threatgrid/ctia/pull/893/files/63b2d7a26805d48429f3bb446e4f62f073a6e744#diff-8613a4405deeeb909529d9ba89de2591R21) junk drawer :-)
https://github.com/threatgrid/ctim/blob/master/src/ctim/schemas/common.cljc#L573

Reply:
Ha! Sure. But I don't know what to do with this information... what do you want/think should to go here?

Reply:
Apparently it's the same schema than [sighting targets](https://github.com/threatgrid/ctim/blob/master/src/ctim/schemas/sighting.cljc#L80).
Thus I would simply use the [sighting target mapping](https://github.com/threatgrid/ctia/blob/master/src/ctia/stores/es/mapping.clj#L227)
``` clojure
:targets em/sighting-target
```